### PR TITLE
.env + LlamaCpp + PDF/CSV + Ingest All

### PR DESCRIPTION
**Pull request by alxspiker**
.env

Added an env file to make configuration easier

LlamaCpp

Added support for LlamaCpp in .env (MODEL_TYPE=LlamaCpp)

PDF/CSV

Added support for PDF and CSV files.

Ingest All

All files in source_documents will automatically get stored in vector store based on their file type when running ingest, no longer need a path argument.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,26 +13,29 @@ In order to set your environment up to run the code here, first install all requ
 pip install -r requirements.txt
 ```
 
+Rename example.env to .env and edit the variables appropriately.
+MODEL_TYPE supports LlamaCpp or GPT4All
```
Comment (imartinez) [inline README.md:17]: It'd be useful to explain in the readme what each variable in .env is used for and the possible values.

File: README.md [hunk 2]
```diff
@@ -13,26 +13,29 @@ In order to set your environment up to run the code here, first install all requ
...
+
 Then, download the 2 models and place them in a folder called `./models`:
 - LLM: default to [ggml-gpt4all-j-v1.3-groovy.bin](https://gpt4all.io/models/ggml-gpt4all-j-v1.3-groovy.bin). If you prefer a different GPT4All-J compatible model, just download it and reference it in `privateGPT.py`.
```
Comment (imartinez) [inline README.md:20]: This line should also be updated to point to .env

File: README.md [hunk 3]
```diff
@@ -13,26 +13,29 @@ In order to set your environment up to run the code here, first install all requ
 pip install -r requirements.txt
 ```
 
+Rename example.env to .env and edit the variables appropriately.
+MODEL_TYPE supports LlamaCpp or GPT4All
+
 Then, download the 2 models and place them in a folder called `./models`:
```
Comment (imartinez) [inline README.md:19]: There is no need to place them in ./models anymore, maybe we should express this differently

**Review by imartinez — COMMENTED** [review]
This PR really improves the functionality in this POC. Thanks a lot for this contribution. Just added a couple comments to make sure the readme is 100% clear and educational.

**Comment by alxspiker** [comment]
Working on it.

File: README.md (hunk 1 above)
Comment (alxspiker) [reply README.md]: Done. Updated to be easier to read too.

File: README.md (hunk 2 above)
Comment (alxspiker) [reply README.md]: Slipped past me, got it

File: README.md (hunk 3 above)
Comment (alxspiker) [reply README.md]: Better wording now?

File: README.md [hunk 4]
```diff
@@ -13,26 +13,34 @@ In order to set your environment up to run the code here, first install all requ
...
-Then, download the 2 models and place them in a folder called `./models`:
-- LLM: default to [ggml-gpt4all-j-v1.3-groovy.bin](https://gpt4all.io/models/ggml-gpt4all-j-v1.3-groovy.bin). If you prefer a different GPT4All-J compatible model, just download it and reference it in `privateGPT.py`.
-- Embedding: default to [ggml-model-q4_0.bin](https://huggingface.co/Pi3141/alpaca-native-7B-ggml/resolve/397e872bf4c83f4c642317a5bf65ce84a105786e/ggml-model-q4_0.bin). If you prefer a different compatible Embeddings model, just download it and reference it in `privateGPT.py` and `ingest.py`.
+Rename example.env to .env and edit the variables appropriately.
+```
+MODEL_TYPE: supports LlamaCpp or GPT4All
+PERSIST_DIRECTORY: is the folder you want your vectorstore in
+LLAMA_EMBEDDINGS_MODEL: Path to your LlamaCpp supported embeddings model
+MODEL_PATH: Path to your GPT4All or LlamaCpp supported LLM
+```
```
Comment (imartinez) [inline README.md:22]: Maybe you could also add a short explanation for MODEL_N_CTX

**Review by imartinez — COMMENTED** [review]
Just a final comment, looking great!

File: README.md (hunk 4 above)
Comment (alxspiker) [reply README.md:23]: Got it

File: README.md [hunk 5]
```diff
@@ -13,26 +13,35 @@ In order to set your environment up to run the code here, first install all requ
...
-Then, download the 2 models and place them in a folder called `./models`:
-- LLM: default to [ggml-gpt4all-j-v1.3-groovy.bin](https://gpt4all.io/models/ggml-gpt4all-j-v1.3-groovy.bin). If you prefer a different GPT4All-J compatible model, just download it and reference it in `privateGPT.py`.
-- Embedding: default to [ggml-model-q4_0.bin](https://huggingface.co/Pi3141/alpaca-native-7B-ggml/resolve/397e872bf4c83f4c642317a5bf65ce84a105786e/ggml-model-q4_0.bin). If you prefer a different compatible Embeddings model, just download it and reference it in `privateGPT.py` and `ingest.py`.
+Rename example.env to .env and edit the variables appropriately.
+```
+MODEL_TYPE: supports LlamaCpp or GPT4All
+PERSIST_DIRECTORY: is the folder you want your vectorstore in
+LLAMA_EMBEDDINGS_MODEL: Path to your LlamaCpp supported embeddings model
+MODEL_PATH: Path to your GPT4All or LlamaCpp supported LLM
+N_CTX: Maximum token limit for both embeddings and LLM models
```
Comment (imartinez) [inline README.md:22]: It should be MODEL_N_CTX I think

Comment (alxspiker) [reply README.md:22]: You are right, sorry working on another branch at the same time, gonna work on one at a time for now.

File: privateGPT.py [hunk 6]
```diff
@@ -2,18 +2,32 @@
...
     retriever = db.as_retriever()
     # Prepare the LLM
     callbacks = [StreamingStdOutCallbackHandler()]
-    llm = GPT4All(model='./models/ggml-gpt4all-j-v1.3-groovy.bin', backend='gptj', callbacks=callbacks, verbose=False)
+    match model_type:
+        case "LlamaCpp":
+            llm = LlamaCpp(model_path=model_path, n_ctx=model_n_ctx, callbacks=callbacks, verbose=False)
+        case "GPT4All":
+            llm = GPT4All(model=model_path, n_ctx=model_n_ctx, backend='gptj', callbacks=callbacks, verbose=False)
+        case _default:
```
Comment (imartinez) [inline privateGPT.py:28]: Shouldn't this be just _ instead of _default? Or does it work the same?

**Review by imartinez — COMMENTED** [review]
Just a final question

File: privateGPT.py (hunk 6 above)
Comment (alxspiker) [reply privateGPT.py:28]: Works the same

Comment (alxspiker) [reply privateGPT.py:28]: Just tested it to make sure, I have always used _default

**Review by imartinez — APPROVED** [review]

**Pull request merged**